process/block/shardblock.go

```diff
@@ -813,6 +813,10 @@ func (sp *shardProcessor) CommitBlock(
 	}
 
 	for i := range finalHeaders {
```

Comment:
blockProcessor should not know about this. we could delete everything before final nonce. Add TODO. Move for i := range finalHeaders to a new function.

Reply:
Added TODO.